File: java/claude/p144_nc.java
import java.sql.*;
import java.util.regex.Pattern;
import java.io.*;

public class p144_nc {
    private static final String DB_PATH = "jdbc:sqlite:/path/to/database.db";
    private static final String PHP_SCRIPT = "/path/to/unsubscribe.php";
    private static final int TIMEOUT_SECONDS = 5;
    private static final Pattern EMAIL_PATTERN = 
        Pattern.compile("^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$");

    public static String validateEmail(String email) {
        String normalized = email.trim().toLowerCase();
        if (EMAIL_PATTERN.matcher(normalized).matches() && normalized.length() <= 254) {
            return normalized;
        }
        return null;
    }

    public static int checkAndUnsubscribe(String email) {
        String normalizedEmail = validateEmail(email);
        if (normalizedEmail == null) {
            return 0;
        }

        try {
            Class.forName("org.sqlite.JDBC");
            Connection conn = DriverManager.getConnection(DB_PATH);
            
            String query = "SELECT 1 FROM users WHERE email = ?";
            PreparedStatement pstmt = conn.prepareStatement(query);
            pstmt.setString(1, normalizedEmail);
            
            ResultSet rs = pstmt.executeQuery();
            boolean exists = rs.next();
            rs.close();
            pstmt.close();
            conn.close();
            
            if (!exists) {
                return 0;
            }

            int result = callPhpScript(normalizedEmail);
            return result;

        } catch (Exception e) {
            return 0;
        }
    }

    private static int callPhpScript(String email) {
        try {
            ProcessBuilder pb = new ProcessBuilder("/usr/bin/php", PHP_SCRIPT, email);
            pb.redirectErrorStream(true);
            Process process = pb.start();

            StringBuilder output = new StringBuilder();
            BufferedReader reader = new BufferedReader(
                new InputStreamReader(process.getInputStream())
            );
            String line;
            while ((line = reader.readLine()) != null) {
                output.append(line);
            }

            boolean finished = process.waitFor(TIMEOUT_SECONDS, java.util.concurrent.TimeUnit.SECONDS);
            
            if (!finished) {
                process.destroyForcibly();
                return 0;
            }

            if (process.exitValue() != 0) {
                return 0;
            }

            return output.toString().trim().equals("0") ? 1 : 0;

        } catch (Exception e) {
            return 0;
        }
    }

    public static void main(String[] args) {
        int result = checkAndUnsubscribe("user@example.com");
        System.out.println(result);
    }
}

File: javascript/claude/p144_nc.js
const { exec } = require("child_process");
const sqlite3 = require("sqlite3").verbose();
const path = require("path");

const DB_PATH = "/path/to/database.db";
const PHP_SCRIPT = "/path/to/unsubscribe.php";
const TIMEOUT_MS = 5000;

function validateEmail(email) {
  const normalized = email.trim().toLowerCase();
  const pattern = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
  if (pattern.test(normalized) && normalized.length <= 254) {
    return normalized;
  }
  return null;
}

function checkAndUnsubscribe(email, callback) {
  const normalizedEmail = validateEmail(email);
  if (!normalizedEmail) {
    return callback(0);
  }

  const db = new sqlite3.Database(DB_PATH, (err) => {
    if (err) {
      return callback(0);
    }

    db.get(
      "SELECT 1 FROM users WHERE email = ?",
      [normalizedEmail],
      (err, row) => {
        db.close();

        if (err || !row) {
          return callback(0);
        }

        const child = require("child_process").spawn("/usr/bin/php", [
          PHP_SCRIPT,
          normalizedEmail,
        ]);

        let output = "";
        let timedOut = false;
        const timer = setTimeout(() => {
          timedOut = true;
          child.kill();
        }, TIMEOUT_MS);

        child.stdout.on("data", (data) => {
          output += data.toString();
        });

        child.on("close", (code) => {
          clearTimeout(timer);

          if (timedOut || code !== 0) {
            return callback(0);
          }

          callback(output.trim() === "0" ? 1 : 0);
        });

        child.on("error", () => {
          clearTimeout(timer);
          callback(0);
        });
      }
    );
  });
}

// Usage
checkAndUnsubscribe("user@example.com", (result) => {
  console.log(result);
});

module.exports = { checkAndUnsubscribe };